client/ayon_deadline/lib.py

```diff
@@ -43,3 +48,368 @@ def get_instance_job_envs(instance) -> "dict[str, str]":
+    OutputDirectory: str = field(
+        default_factory=partial(DeadlineIndexedVar, "OutputDirectory"))
+
+    AssetDependency: str = field(
+        default_factory=partial(DeadlineIndexedVar, "AssetDependency"))
+
+    TileJob: bool = field(default=False)
+    TileJobFrame: int = field(default=0)
+    TileJobTilesInX: int = field(default=0)
+    TileJobTilesInY: int = field(default=0)
+    TileJobTileCount: int = field(default=0)
+
+    MaintenanceJob: bool = field(default=False)
+    MaintenanceJobStartFrame: int = field(default=0)
+    MaintenanceJobEndFrame: int = field(default=0)
+
+
+@dataclass
+class AYONDeadlineJobInfo(DeadlineJobInfo):
+    """Contains additional AYON variables from Settings for internal logic."""
+
+    # AYON custom fields used for Settings
+    UsePublished: Optional[bool] = field(default=None)
+    UseAssetDependencies: Optional[bool] = field(default=None)
+    UseWorkfileDependency: Optional[bool] = field(default=None)
+
+    MachineList: Optional[str] = field(
+        default=None)  # Default blank (comma-separated list)
+    MachineListDeny: Optional[bool] = field(default=None)
+
+    AdditionalJobInfo: Optional[str] = field(default=None)  # Default: blank
+    AdditionalPluginInfo: Optional[str] = field(default=None)  # Default: blank
+
+    def serialize(self):
+        """Return all data serialized as dictionary.
+
+        Returns:
+            OrderedDict: all serialized data.
+
+    """
+        def filter_data(a, v):
+            if isinstance(v, (DeadlineIndexedVar, DeadlineKeyValueVar)):
+                return False
+            if v is None:
+                return False
+            return True
+
+        serialized = asdict(self)
+        serialized = {
+            k: v for k, v in serialized.items()
+            if filter_data(k, v)
+        }
+
+        # Custom serialize these attributes
+        for attribute in [
+            self.EnvironmentKeyValue,
+            self.ExtraInfo,
+            self.ExtraInfoKeyValue,
+            self.TaskExtraInfoName,
+            self.OutputFilename,
+            self.OutputFilenameTile,
+            self.OutputDirectory,
+            self.AssetDependency
+        ]:
+            serialized.update(attribute.serialize())
+
+        return serialized
+
+    @classmethod
+    def from_dict(cls, data: Dict) -> 'JobInfo':
+
+        def capitalize(key):
+            """Transform AYON looking variables from Settings to DL looking.
+
+            AYON uses python like variable names, eg use_published, DL JobInfo
+            uses capitalized, eg. UsePublished.
+            This method does the conversion based on this assumption.
+            """
+            words = key.split("_")
+            return "".join(word.capitalize() for word in words)
+
+        # Filter the dictionary to only include keys that are fields in the dataclass
+        capitalized = {capitalize(k): v for k, v in data.items()}
```

Comment:
This type of magical conversion makes it really hard to find and track where these attributes are coming from in the codebase, because suddenly you can't just `CTRL+F` to see all occurrences of a particular key?

In this case I think we're better off just hardcoded the expected 'translations' or not convert them at all?

It's extra confusing because `UsePublished` - that's not a deadline job info key at all, right?

Reply:
Yeah, I could change it to hardcoded mapping, the idea is that we dont have field like `ChunkSize` is Settings, there it follows python formatting. There is not too many of those, one would need to remember if adding new field to Settings they must add them to mapping.

Reply:
Yup - I think we're better off doing an explicit mapping. Unless those are 100s of entries?

Reply:
No now, changed to explicit.